dynamic/parameterize/args.go

```diff
@@ -72,14 +74,27 @@ func ParseArgs(ctx context.Context, a []string) (Args, error) {
 	var indexDocOutDir string
 	var includes []string
 	var providerName string
+
+	// If --help is included in `a`, then `RunE` will never be executed.
+	//
+	// We detect that and make sure we return an error instead of continuing.
+	var cmdWasRun bool
+
 	cmd := cobra.Command{
 		Use: "./local | remote version",
 		RunE: func(cmd *cobra.Command, a []string) error {
+			cmdWasRun = true
 			var err error
 			args, err = parseArgs(cmd.Context(), a, fullDocs, upstreamRepoPath, indexDocOutDir, includes, providerName)
 			return err
 		},
-		Args: cobra.RangeArgs(1, 2),
+		Args: func(cmd *cobra.Command, args []string) error {
+			err := cobra.RangeArgs(1, 2)(cmd, args)
+			if err == nil {
+				return nil
+			}
+			return status.Error(codes.InvalidArgument, err.Error())
```

Comment:
Does this strictly need to be a grpc error?
And could we rewrite this to the more common "if err != nil" or is there some subtlety I'm missing?

Reply:
It doesn't, but it is more "correct". I live in hope that the CLI will start recognizing these error codes and make the display look better.